HACKING.md

```diff
@@ -4,70 +4,84 @@ Hacking on `snapd` is fun and straightforward. The code is extensively unit
 tested and we use the [spread](https://github.com/snapcore/spread)
 integration test framework for the integration/system level tests.
 
-## Development
+## Setting up
 
 ### Supported Ubuntu distributions
 
 Ubuntu 18.04 LTS or later is recommended for `snapd` development.
+Usually, the latest LTS would be the best choice.
 
-If you want to build or test on older versions of Ubuntu, additional steps
+>If you want to build or test on older versions of Ubuntu, additional steps
 may be required when installing dependencies.
 
 
 ### Supported Go version
 
-Go 1.13 or later is required to build `snapd`.
+[Go 1.13](https://pkg.go.dev/go@go1.13) or later is required to build `snapd`.
 
-If you need to build older versions of snapd, please have a look at the file
-`debian/control` to find out what dependencies were needed at the time
+>If you need to build older versions of snapd, please have a look at the file
+[debian/control](debian/control) to find out what dependencies were needed at the time
 (including which version of the Go compiler).
 
-### Setting up your build environment
+### Setting up build environment
 
-If your Go environment (e.g. `GOPATH`) is already configured, you should skip
-this step. The Go environment setup can be added to your shell login script
-(e.g. `~/.bashrc` for bash) for automatic setup (after a terminal restart).
+`snapd` is published and maintained as a [Go module](https://go.dev/doc/modules/managing-source). 
+A module is a collection of [Go packages](https://go.dev/ref/spec#Packages) 
+stored in a file tree with a [go.mod](go.mod) file at its root. 
+The [go.mod](go.mod) file defines the module’s _module path_, 
+which is also the import path used for the root directory, 
+and its _dependency requirements_. 
+All this allows maintaining `snapd` outside of the strict hierarchy of `GOPATH`, 
+that was mandatory before the release of [Go 1.13](https://pkg.go.dev/go@go1.13)
+(see [build @Go1.10](https://pkg.go.dev/go/build@go1.10)).  
```

Comment:
I wasn't, and the Go documentation wasn't really straightforward on this topic. On contrary, in this area the documentation was very quite confusing for me.   

@MiguelPires, I think that this document is not for the elders. Though, I think not even all elders know all the details - simply because they don't need it. The fresh ones (who actually start from reading this document) do need this and also proper links to have more details that just one sentence in order to understand how this things work (in my opinion of course).

By the way, even Go documentation is written by Go pros, who not always can see what are the difficulties of the newbies.